### PR TITLE
anbox: drop kernel modules package

### PR DESCRIPTION
**Pull request by Mindavi**
All builds are broken:
- 4.14
- 4.19
- 5.4

https://hydra.nixos.org/eval/1761072?filter=anbox&compare=1760998&full=#tabs-still-fail

Let's just drop it.

This leaves the anbox module possibly in a broken-ish state,
but I'm not sure what to do about it.

cc @samueldr @edwtjo @cidkidnix @mzabani @Atemu @mvnetbiz 

Hopefully someone of you can tell me if I break everything with this. Maybe there should be more/less things to drop from the module?

A test for the module would've been really nice to verify this doesn't break anything.

ZHF: https://github.com/NixOS/nixpkgs/issues/172160

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/virtualisation/anbox.nix [hunk 1]
```diff
@@ -73,9 +73,6 @@ in
 
     environment.systemPackages = with pkgs; [ anbox ];
 
-    boot.kernelModules = [ "ashmem_linux" "binder_linux" ];
```
Comment (Atemu) [inline nixos/modules/virtualisation/anbox.nix:76]: We still want the modules to be loaded I think? I don't use anbox anymore.

**Review by Atemu — CHANGES_REQUESTED** [review]
Newer kernels (5.0 upwards) actually include these modules, so the external kmod isn't necessary anymore. 

If they're broken on older kernels anyways, might as well drop the package. It won't be needed in the future either.

File: nixos/modules/virtualisation/anbox.nix (hunk 1 above)
Comment (cidkidnix) [reply nixos/modules/virtualisation/anbox.nix:76]: There is no "modules" for anbox with the kernel config that got added, they're built-in to kernel now

Comment (Mindavi) [reply nixos/modules/virtualisation/anbox.nix:76]: Yeah, that's what I gathered from looking at other PRs related to this as well.

**Review by davidak — APPROVED** [review]
Change looks good

File: nixos/modules/virtualisation/anbox.nix (hunk 1 above)
Comment (Atemu) [reply nixos/modules/virtualisation/anbox.nix:76]: I know but they still need to be loaded before anbox can work. I don't think they're loaded automatically unless we declare NixOS to do so.

Comment (Mindavi) [reply nixos/modules/virtualisation/anbox.nix:76]: Hopefully an anbox user can support with that.

**Pull request merged**